Content.Client/UserInterface/Systems/Radial/Controls/RadialContainer.xaml.cs

```diff
@@ -0,0 +1,403 @@
+    /// </summary>
+    public Action<RadialContainer>? OnClose;
+
+    public Action<RadialContainer>? OnAttached;
+    public Action<RadialContainer>? OnDetached;
+
+    /// <summary>
+    /// Radial item size, when cursor was focused on button
+    /// </summary>
+    public float FocusSize { get; set; } = 64f;
+
+    /// <summary>
+    /// Normal radial item size, when cursor not focused
+    /// </summary>
+    public float NormalSize { get; set; } = 50f;
+
+    /// <summary>
+    /// Items moving animation time, when radial was opened
+    /// </summary>
+    public float MoveAnimationTime { get; set; } = 0.3f;
+
+    /// <summary>
+    /// Item focus sizing animation, when cursor focused and leaved
+    /// </summary>
+    public float FocusAnimationTime { get; set; } = 0.25f;
+
+    /// <summary>
+    /// Show or not item action text on radial menu
+    /// </summary>
+    public bool ActionVisible { get; set; } = true;
+
+    /// <summary>
+    /// Used for UIController! Radial menu position offset
+    /// </summary>
+    public float VerticalOffset { get; set; } = 0.0f;
+
+    /// <summary>
+    /// Used for UIController! Radius, where is container was available
+    /// </summary>
+    public float VisionRadius { get; set; } = 10.0f;
+
+    /// <summary>
+    /// Used for automatic set RadialItem.Controller texture
+    /// </summary>
+    public string ItemBackgroundTexturePath { get; set; }
+
+    public EntityUid? AttachedEntity
+    {
+        get => _attachedEntity;
+        set
+        {
+            if (value == EntityUid.Invalid || value == null)
+                OnDetached?.Invoke(this);
+            else
+                OnAttached?.Invoke(this);
+
+            _attachedEntity = value;
+        }
+    }
+
+    public RadialContainer()
+    {
+        RobustXamlLoader.Load(this);
+        IoCManager.InjectDependencies(this);
+
+        var background = Theme.Path.ToString();
+        ItemBackgroundTexturePath = background += "SlotBackground";
+    }
+
+    /// <summary>
+    /// Open container
+    /// </summary>
+    public void Open()
+    {
+        if (_isOpened)
+            return;
+
+        _isOpened = !_isOpened;
+
+        UpdateButtons();
+    }
+
+    /// <summary>
+    /// Open container with position
+    /// </summary>
+    /// <param name="position"></param>
+    public void Open(Vector2 position)
+    {
+        if (_isOpened)
+            return;
+
+        _isOpened = !_isOpened;
+
+        LayoutContainer.SetPosition(this, position);
+        UpdateButtons();
+    }
+
+    /// <summary>
+    /// Open container on center screen
+    /// </summary>
+    public void OpenCentered()
+    {
+        if (_isOpened)
+            return;
+
+        _isOpened = !_isOpened;
+
+        if (Parent == null)
+            return;
+
+        LayoutContainer.SetPosition(this, (Parent.Size/2) - (Size/2));
+        UpdateButtons();
+    }
+
+    public void OpenCenteredLeft() => OpenCenteredAt(new Vector2(0.25f, 0.5f));
+
```

Comment:
As mentioned in the previous PR a version attached to entities should be split out as now it gets diluted between the two.